### PR TITLE
feat: upgrade to cdk v2

### PR DESCRIPTION
**Pull request by corymhall**
This PR upgrades delivlib to use CDK v2. It also migrates the tests to
use the `assertions` library.

BREAKING CHANGE: delivlib now requires cdk v2

re #1841

-----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: lib/__tests__/registry-sync/mirror-source.test.ts [hunk 1]
```diff
@@ -79,9 +78,15 @@ describe('RegistryImageSource', () => {
...
+      const app = new App();
+      const stack = new Stack(app, 'Default', {
+        env: {
+          account: '111111111111',
+          region: 'us-east-1',
+        },
+      });
       const ecrRegistry = 'myregistry';
-      const source = MirrorSource.fromDirectory(path.join(__dirname, 'docker-asset'), 'myrepository');
+      const source = MirrorSource.fromDir(path.join(__dirname, 'docker-asset'), 'myrepository');
```
Comment (corymhall) [inline lib/__tests__/registry-sync/mirror-source.test.ts:89]: `fromDirectory` is deprecated.

File: lib/__tests__/pipeline.test.ts [hunk 2]
```diff
@@ -92,60 +95,61 @@ test('can add arbitrary shellables with different artifacts', () => {
...
       },
       {
-        Name: 'Publish',
         Actions: [
-          {
-            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild' },
-            InputArtifacts: [{ Name: 'Artifact_TestStackPipelineSecondStep91726C11' }],
+          Match.objectLike({
+            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild', Version: '1' },
+            InputArtifacts: [{ Name: 'Artifact_c81eddcbe9657bd312a728fb13df77bc09f9a519b4' }],
```
Comment (corymhall) [inline lib/__tests__/pipeline.test.ts:144]: The artifact name change is what we get when we go from `node.uniqueId` => `node.addr`

File: lib/__tests__/expected.yml [hunk 3]
```diff
@@ -568,7 +234,7 @@ Resources:
                 - Name: Artifact_Build_Build
               Name: TestHelloLinux
               OutputArtifacts:
-                - Name: Artifact_delivlibtestCodeCommitPipelineHelloLinux01786D9D
+                - Name: Artifact_c883e6647f907b1eb255846397acb348c18b48b3a2
```
Comment (corymhall) [inline lib/__tests__/expected.yml:237]: The artifact name change is what we get when we go from `node.uniqueId` => `node.addr`

File: lib/__tests__/chime-notifier.test.ts [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
 import { ChimeNotifier } from '../../lib';
 import { codePipeline, handler } from '../../lib/chime-notifier/handler/notifier-handler';
-// eslint-disable-next-line @typescript-eslint/no-require-imports
-const https = require('https');
-import '@monocdk-experiment/assert/jest';
 
-jest.mock('https');
-
-const mockHttpsWrite = jest.fn();
-(https as any).request = jest.fn((_url, _options, cb) => {
```
Comment (corymhall) [inline lib/__tests__/chime-notifier.test.ts:15]: One of the dependency updates did not like `(https as any)`

File: lib/__tests__/change-control-lambda/handler.test.ts [hunk 5]
```diff
@@ -14,12 +10,22 @@ import * as timeWindow from '../../change-control-lambda/time-window';
...
+    CodePipeline: jest.fn(() => {
+      return {};
+    }).mockName('AWS.CodePipeline'),
+  };
+});
 jest.mock('../../change-control-lambda/disable-transition');
 jest.mock('../../change-control-lambda/time-window');
 
 const mockGetObject = jest.fn().mockName('AWS.S3.getObject');
-(AWS as any).S3 = jest.fn(() => ({ getObject: mockGetObject })).mockName('AWS.S3') as any;
```
Comment (corymhall) [inline lib/__tests__/change-control-lambda/handler.test.ts:22]: One of the dependency updates did not like `(AWS as any)` so I updated the test.

File: lib/__tests__/canary.test.ts [hunk 6]
```diff
@@ -66,45 +68,13 @@ test('correctly creates canary', () => {
...
-                          Ref: 'AssetParameters3d34b07ba871989d030649c646b3096ba7c78ca531897bcdb0670774d2f9d3e4S3VersionKeyF3F83F76',
-                        },
-                      ],
-                    },
-                  ],
-                },
-              ],
-            ],
-          },
+          Value: '3d34b07ba871989d030649c646b3096ba7c78ca531897bcdb0670774d2f9d3e4.zip',
```
Comment (corymhall) [inline lib/__tests__/canary.test.ts:77]: This is probably going to happen in `aws-cdk` when we start running tests for v2.

**Review by corymhall — COMMENTED** [review]
Couple of explanations.

**Comment by corymhall** [comment]
> I'm okay with it, but needs a major version bump

Shouldn't the major version bump happen automatically because of the `BREAKING CHANGE`?

**Review by rix0rrr — APPROVED** [review]

**Pull request merged**